Sources/Interaction/Widgets/ImageCroppingRegionsRepresentation/index.js

```diff
@@ -0,0 +1,384 @@
+      case Orientation.XZ:
+        verts.set([xBMin, slicePos, zBMin], 0);
+        verts.set([xPLower, slicePos, zBMin], 3);
+        verts.set([xPLower, slicePos + 0.01, zPLower], 6);
+        verts.set([xBMin, slicePos, zPLower], 9);
+        verts.set([xPUpper, slicePos + 0.01, zBMin], 12);
+        verts.set([xBMax, slicePos, zBMin], 15);
+        verts.set([xBMax, slicePos + 0.01, zPLower], 18);
+        verts.set([xPUpper, slicePos, zPLower], 21);
+        verts.set([xPUpper, slicePos + 0.01, zPUpper], 24);
+        verts.set([xBMax, slicePos, zPUpper], 27);
+        verts.set([xBMax, slicePos, zBMax], 30);
+        verts.set([xPUpper, slicePos, zBMax], 33);
+        verts.set([xBMin, slicePos, zPUpper], 36);
+        verts.set([xPLower, slicePos, zPUpper], 39);
+        verts.set([xPLower, slicePos + 0.01, zBMax], 42);
+        verts.set([xBMin, slicePos, zBMax], 45);
+
+        centerVerts.set([xPLower, slicePos, zPLower], 0);
+        centerVerts.set([xPUpper, slicePos, zPLower], 3);
+        centerVerts.set([xPUpper, slicePos, zPUpper], 6);
+        centerVerts.set([xPLower, slicePos, zPUpper], 9);
+        break;
+      case Orientation.XY:
+        verts.set([xBMin, yBMin, slicePos], 0);
+        verts.set([xPLower, yBMin, slicePos], 3);
+        verts.set([xPLower, yPLower, slicePos + 0.01], 6);
+        verts.set([xBMin, yPLower, slicePos], 9);
+        verts.set([xPUpper, yBMin, slicePos + 0.01], 12);
+        verts.set([xBMax, yBMin, slicePos], 15);
+        verts.set([xBMax, yPLower, slicePos + 0.01], 18);
+        verts.set([xPUpper, yPLower, slicePos], 21);
+        verts.set([xPUpper, yPUpper, slicePos + 0.01], 24);
+        verts.set([xBMax, yPUpper, slicePos], 27);
+        verts.set([xBMax, yBMax, slicePos], 30);
+        verts.set([xPUpper, yBMax, slicePos], 33);
+        verts.set([xBMin, yPUpper, slicePos + 0.01], 36);
+        verts.set([xPLower, yPUpper, slicePos], 39);
+        verts.set([xPLower, yBMax, slicePos + 0.01], 42);
+        verts.set([xBMin, yBMax, slicePos], 45);
+
+        centerVerts.set([xPLower, yPLower, slicePos], 0);
+        centerVerts.set([xPUpper, yPLower, slicePos], 3);
+        centerVerts.set([xPUpper, yPUpper, slicePos], 6);
+        centerVerts.set([xPLower, yPUpper, slicePos], 9);
+        break;
+      default:
+      // noop
+    }
+
+    // This requires updating whenever geometry updates
+    updateCenterOpacity();
+
+    model.regionPolyData.modified();
+    model.centerRegionPolyData.modified();
+    publicAPI.invokePlanesPositionChanged();
+  };
+
+  publicAPI.getBounds = () => model.initialBounds;
+
+  publicAPI.buildRepresentation = () => {
+    if (model.renderer) {
+      if (!model.placed) {
+        model.validPick = 1;
+        model.placed = 1;
+      }
+      publicAPI.modified();
+    }
+  };
+
+  publicAPI.setProperty = (property) => {
+    model.actor.setProperty(property);
+  };
+
+  // invoke the constructor after setting up public methods
+  constructor();
```

Comment:
Why don't you inline the constructor content right there?

Reply:
Having the constructor code at the top was just a preference.

Reply:
What about having a `initialize(publicAPI, model)` outside of the function, at the very top of the file?

Reply:
either way, I'm fine, I was just curious regarding that closure.

Reply:
I've been treating closures inside the main method as private methods that can access private variables in the upper scope of the closures. I don't have any private variables in this class, but I still wrote the code like that for consistency. :P

Reply:
A function inside a module is private too...

Reply:
Agreed. I put those private methods alongside private variables (like in ImageCropRegionsWidget) in the main function, since I opted to not put the variables in the module scope.

I can still go ahead and make changes if you'd like, since I'm not partial to any one approach and it's just what convention we want to go with at this point.

Reply:
I think we use helper functions at the module scope sometime when needed as private function that only needs to be define once.

Since I like you comment ` invoke after setting up public methods`, I can see the initialization process done in a function. But I would put it at the module level with a name like `initialize(publicAPI, model)` or `setup(publicAPI, model)` or something else. But I don't like `constructor()` as it is like an ES6 class which we are not and that can be confusing.

Reply:
Ah, agreed with the ES6 class naming clash there. I'm thinking of moving the representation construction to buildRepresentation as per the comment I made [here](https://github.com/Kitware/vtk-js/pull/496#discussion_r160814080), which would remove the need to have a separate `initialize(publicAPI, model)` function.